R/dataframe-frame.R

```diff
@@ -499,3 +499,41 @@ dataframe__sort <- function(
   )$collect(`_eager` = TRUE) |>
     wrap()
 }
+
+#' Get number of chunks used by the ChunkedArrays of this DataFrame
+#'
+#' @param strategy Return the number of chunks of the `"first"` column, or
+#' `"all"` columns in this DataFrame.
+#'
+#' @return An integer vector.
+#' @examples
+#' df <- pl$DataFrame(
+#'   a = c(1, 2, 3, 4),
+#'   b = c(0.5, 4, 10, 13),
+#'   c = c(TRUE, TRUE, FALSE, TRUE)
+#' )
+#'
+#' df$n_chunks()
+#' df$n_chunks(strategy = "all")
+dataframe__n_chunks <- function(strategy = c("first", "all")) {
+  wrap({
+    strategy <- arg_match0(strategy, values = c("first", "all"))
+    if (strategy == "first") {
+      self$`_df`$n_chunks()
+    } else {
+      vapply(as.list(self, as_series = TRUE), \(x) x$n_chunks(), FUN.VALUE = integer(1))
```

Comment:
Two reasons:
1. it directly returns a vector, so no need to call `unlist()` at the end
2. it ensures the output of each iteration is as expected (integer here). If `n_chunks()` changes later, it would probably be caught elsewhere anyway, but I took the habit of being explicit about the output type when I can (it's also easier to read IMO).

Reply:
Thank you. I am convinced.